x/bridge/observer/osmosis/osmosis_test.go

```diff
@@ -0,0 +1,323 @@
+	go ots.ts.Start(t)
+	ots.o.Start(ctx)
+}
+
+func (ots *OsmosisTestSuite) Stop(t *testing.T, ctx context.Context) {
+	ots.o.Stop(ctx)
+	ots.hs.Close()
+	ots.ts.Close(t)
+}
+
+var upgrader = websocket.Upgrader{}
+
+func readNewBlockEvent(t *testing.T, path string) coretypes.ResultEvent {
+	dataStr, err := os.ReadFile(path)
+	require.NoError(t, err)
+	result := coretypes.ResultEvent{}
+	err = cmtjson.Unmarshal([]byte(dataStr), &result)
+	require.NoError(t, err)
+	return result
+}
+
+func readTxSearch(t *testing.T, path string) coretypes.ResultTxSearch {
+	dataStr, err := os.ReadFile(path)
+	require.NoError(t, err)
+	result := coretypes.ResultTxSearch{}
+	err = json.Unmarshal([]byte(dataStr), &result)
+	require.NoError(t, err)
+	return result
+}
+
+func success(t *testing.T) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		switch r.Method {
+		case http.MethodGet:
+			c, err := upgrader.Upgrade(w, r, nil)
+			require.NoError(t, err)
+			defer c.Close()
+			newBlock := readNewBlockEvent(t, "./test_events/new_block_event.json")
+			newBlockResp := cmtrpctypes.NewRPCSuccessResponse(
+				cmtrpctypes.JSONRPCIntID(1),
+				newBlock,
+			)
+			newBlockRaw, err := json.Marshal(newBlockResp)
+			require.NoError(t, err)
+			err = c.WriteMessage(1, newBlockRaw)
+			require.NoError(t, err)
+		case http.MethodPost:
+			blockResults := readTxSearch(t, "./test_events/tx_search.json")
+			blockResultsResp := cmtrpctypes.NewRPCSuccessResponse(
+				cmtrpctypes.JSONRPCIntID(0),
+				blockResults,
+			)
+			blockResultsRaw, err := json.Marshal(blockResultsResp)
+			require.NoError(t, err)
+			_, err = w.Write(blockResultsRaw)
+			require.NoError(t, err)
+		default:
+			t.Fatal("Unexpected request method", r.Method)
+		}
+	}
+}
+
+// TestSignalInboundTransfer verifies calling SignalInboundTransfer
+// results in Tx being signed and sent to the chain
+func TestSignalInboundTransfer(t *testing.T) {
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second)
+	defer cancel()
+	ots := NewOsmosisTestSuite(t, ctx)
+
+	expectedAcc := authtypes.NewBaseAccount(Addr1, nil, 1, 2)
+	expReq1 := &authtypes.QueryAccountRequest{
+		Address: Addr1.String(),
+	}
+	m, err := expectedAcc.Marshal()
+	require.NoError(t, err)
+	expResp1 := &authtypes.QueryAccountResponse{
+		Account: &codectypes.Any{
+			Value: m,
+		},
+	}
+	ots.ts.AccServer.
+		EXPECT().
+		Account(gomock.Any(), expReq1).
+		Times(1).
+		Return(expResp1, nil)
+
+	keyring := keyring.NewInMemory(app.GetEncodingConfig().Marshaler)
+	_, err = keyring.NewAccount(
+		osmosis.ModuleNameClient,
+		Mnemonic1,
+		"",
+		types.FullFundraiserPath,
+		hd.Secp256k1,
+	)
+	require.NoError(t, err)
+
+	in := observer.Transfer{
+		SrcChain: observer.ChainIdBitcoin,
+		DstChain: observer.ChainIdOsmosis,
+		Id:       "deadbeef",
+		Height:   42,
+		Sender:   Addr1.String(),
+		To:       BtcAddr,
+		Asset:    "btc",
+		Amount:   math.NewUint(10),
+	}
```

Comment:
An inbound transfer to osmosis cant have a to address be a BtcAddr right? I would expect this to fail at the validateBasic level